src/main/scala/io/citrine/lolo/trees/multitask/MultiTaskTrainingNode.scala

```diff
@@ -2,171 +2,207 @@ package io.citrine.lolo.trees.multitask
 
 import io.citrine.lolo.PredictionResult
 import io.citrine.lolo.linear.GuessTheMeanLearner
+import io.citrine.lolo.trees.classification.ClassificationTrainingLeaf
 import io.citrine.lolo.trees.regression.RegressionTrainingLeaf
-import io.citrine.lolo.trees.splits.{MultiTaskSplitter, NoSplit}
-import io.citrine.lolo.trees.{InternalModelNode, ModelNode, TrainingLeaf}
+import io.citrine.lolo.trees.splits.{MultiTaskSplitter, NoSplit, Split}
+import io.citrine.lolo.trees.{InternalModelNode, ModelNode, TrainingNode}
 import io.citrine.random.Random
 
 import scala.collection.mutable
 
-/**
-  * Node in a multi-task training tree, which can produce nodes for its model trees.
-  * Splits are chosen using a MultiTaskSplitter, which considers the sum impurity decrease across all tasks.
-  *
-  * @param inputs data on which to select splits and form models
-  * @param numFeatures to randomly select from at each split (negative values indicate that all features should be considered)
-  * @param maxDepth to grow the tree to
-  * @param minInstances minimum number of training instances per leaf
-  * @param randomizePivotLocation whether to generate splits randomly between the data points
-  * @param rng random number generator, for reproducibility
-  */
-class MultiTaskTrainingNode(
-    inputs: Seq[(Vector[AnyVal], Array[AnyVal], Double)],
-    numFeatures: Int,
-    maxDepth: Int,
-    minInstances: Int,
-    randomizePivotLocation: Boolean = false,
-    rng: Random
+case class MultiTaskTrainingNode(
+    trainingData: Seq[(Vector[AnyVal], Array[AnyVal], Double)],
+    labelWiseInstructions: Seq[MultiTaskLabelInstruction],
```

Comment:
`MultiTaskTrainingNode` presents additional complications because the tree traversal logic is label-dependent. A split might exist, but if there are no more training rows for some label at this node, then we must terminate and make a prediction.
The solution here is to include a sequence of instructions saying what to do for each label -- take the split, evaluate a given leaf model, go left, or go right. This is how we can smuggle the label-wise models in to various points on the tree. But there's a flaw in this approach that I have not been able to get around -- there's nothing fundamental preventing an inconsistency between the instructions and the child nodes. For example, what if the label instruction is `DoSplit()` but the left/right node options are `None`?
In practice, the logic of the builder method prevents this from happening. But it's still a flaw.

Reply:
That could work. It means that the split info will be duplicated several times. I'll try it out.

Reply:
Changed it. There's some duplication but I think it's better overall.